_pytest/skipping.py

```diff
@@ -155,17 +155,17 @@ def getexplanation(self):
 @hookimpl(tryfirst=True)
 def pytest_runtest_setup(item):
     # Check if skip or skipif are specified as pytest marks
-
+    item._skipped_by_mark = False
```

Comment:
Should we make this part of the public API? Often users want to know if a test has been skipped in some hook, and the only way to do it currently is using the awkward:

```python
was_skipped = hasattr(item, "_evalskip") and item._evalskip.istrue()
```

We might as well make it `item.skipped_by_mark`.

Reply:
While on that, it might also be useful to have the reason attached tn it. If so, perhaps we should use a frozen `attrs` instance which also contains the reason? This makes it possible to add more information to it in the future:

```python
item.skip_outcome.skipped == True
item.skip_outcome.reason == "skipped for some reason"
```

Reply:
im not going to introduce a new api in a refactoring/fixup attempt,
thats for a follow-up - please open a issue for better communicating outcomes

Reply:
I understand the sentiment, but by renaming it to `_skipped_by_mark` you will break code which currently looks at this attribute to see if a test was skipped (for example IDEs, see #1613). So IDE and plugin authors which use this to check if a test was skipped will now have to test for `_evalskip` and `_skipped_by_mark` to comply. And *then* when a new version with a public API comes around, we will have another way to check for that.

Just pointing out we will bring a lot of pain to those people. Perhaps we can brainstorm a little on how the "skip" plugin decorates items with skip information?

Reply:
its a private attribute and stays a private attribute
so im not going to protect them too much from that use, thats what they should have  CI for